lib/iris/tests/test_basic_maths.py

```diff
@@ -685,12 +685,12 @@ def setUp(self):
         self.data_1u = np.array([[9, 9, 9], [8, 8, 8]], dtype=np.uint64)
         self.data_2u = np.array([[3, 3, 3], [2, 2, 2]], dtype=np.uint64)
 
-        self.cube_1f = iris.cube.Cube(self.data_1f)
-        self.cube_2f = iris.cube.Cube(self.data_2f)
-        self.cube_1i = iris.cube.Cube(self.data_1i)
-        self.cube_2i = iris.cube.Cube(self.data_2i)
-        self.cube_1u = iris.cube.Cube(self.data_1u)
-        self.cube_2u = iris.cube.Cube(self.data_2u)
+        self.cube_1f = iris.cube.Cube(self.data_1f.copy())
+        self.cube_2f = iris.cube.Cube(self.data_2f.copy())
+        self.cube_1i = iris.cube.Cube(self.data_1i.copy())
+        self.cube_2i = iris.cube.Cube(self.data_2i.copy())
+        self.cube_1u = iris.cube.Cube(self.data_1u.copy())
+        self.cube_2u = iris.cube.Cube(self.data_2u.copy())
```

Comment:
Added these copy's because I concluded that previously the in-place tests were not very good.  Under Iris2.4, we have:
```python
import iris
import numpy as np

data_1f = np.array([[9, 9, 9], [8, 8, 8]], dtype=np.float64)
data_2f = np.array([[3, 3, 3], [2, 2, 2]], dtype=np.float64)

cube_1f = iris.cube.Cube(data_1f)
cube_2f = iris.cube.Cube(data_2f)

cube_1f += cube_2f
data_1f += data_2f

print(data_1f)
print(cube_1f.data is data_1f)
```
gives:
```
[[15. 15. 15.]
 [12. 12. 12.]]
True
```